beetsplug/embedart.py

```diff
@@ -60,22 +60,59 @@ def commands(self):
         embed_cmd.parser.add_option(
             u'-f', u'--file', metavar='PATH', help=u'the image file to embed'
         )
+        embed_cmd.parser.add_option(
+            u"-y", u"--yes", action="store_true", help=u"skip confirmation"
+        )
         maxwidth = self.config['maxwidth'].get(int)
         compare_threshold = self.config['compare_threshold'].get(int)
         ifempty = self.config['ifempty'].get(bool)
 
+        def _confirmation(items, opts, fmt, prompt):
+            # Confirm artwork changes to library items.
+                if not opts.yes:
+                    # Prepare confirmation with user.
+                    print_()
+
+                    # Show all the items.
+                    for item in items:
+                        print_(format(item, fmt))
+
+                    # Confirm with user.
+                    if not ui.input_yn(prompt, True):
+                        return False
+                return True
+
         def embed_func(lib, opts, args):
             if opts.file:
                 imagepath = normpath(opts.file)
                 if not os.path.isfile(syspath(imagepath)):
                     raise ui.UserError(u'image file {0} not found'.format(
                         displayable_path(imagepath)
                     ))
-                for item in lib.items(decargs(args)):
+
+                items = lib.items(decargs(args))
+
+                # Confirm with user.
+                fmt = u'$albumartist - $album - $title'
+                prompt = u'Modify artwork for %i file%s (y/n)?' % \
+                             (len(items), 's' if len(items) > 1 else '')
+                if not _confirmation(items, opts, fmt, prompt):
+                    return
+
+                for item in items:
                     art.embed_item(self._log, item, imagepath, maxwidth, None,
                                    compare_threshold, ifempty)
             else:
-                for album in lib.albums(decargs(args)):
+                items = lib.albums(decargs(args))
+
+                # Confirm with user.
+                fmt = u'$albumartist - $album'
+                prompt = u'Modify artwork for %i album%s (y/n)?' % \
+                         (len(items), 's' if len(items) > 1 else '')
+                if not _confirmation(items, opts, fmt, prompt):
+                    return
```

Comment:
Funny you mention that here, but that is exactly why I created the _confirmation method. They are very similar, sure but there are subtle differences (items vs. albums for instance and differences in the prompt and format strings. Like the comment above, I really don't know how I would increase the reusability here.